bin/git-bulk

```diff
@@ -0,0 +1,159 @@
+    read -n 1 -r </dev/tty; echo
+    if [[ $REPLY =~ ^[Yy]$ ]]; then atomicExecution; fi
+  else
+     atomicExecution
+  fi
+}
+
+# atomic git command execution with log
+function atomicExecution () {
+  echo "${bldred}->${reset} executing ${invers}git $gitcommand${reset}" && git $gitcommand
+}
+
+# check if the passed command is known as a core git command
+function checkGitCommand () {
+  if git help -a | cat | grep -o -q "\b${corecommand}\b"; then
+    echo "Core command \"$corecommand\" accepted."
+  else
+    if git config --get-regexp alias | grep -o -q "\.${corecommand} "; then
+      echo "Alias ${corecommand} accepted."
+    else
+      usage && echo "error: unknown GIT command: $corecommand" && exit 1
+    fi
+  fi
+}
+
+# check if workspace name is registered
+function checkWSName () {
+  found=0
+  while read workspace; do
+    rwsname=$(echo $workspace | cut -f1 -d' ' | cut -f2 -d'.')
+    if [[ $rwsname == $wsname ]]; then return; fi
+  done <<< "$(echo "$(listall)")"
+  # when here the ws name was not found
+  usage && echo "error: unknown workspace name: $wsname" && exit 1 
+}
+
+# detects the wsname of the current directory
+function wsnameToCurrent () {
+  currdir=$PWD
+  while read workspace; do
+    if [ -z "$workspace" ]; then continue; fi
+    rwsdir=$(echo $workspace | grep -o ' /.*')
+    rwsname=$(echo $workspace | cut -f1 -d' ' | cut -f2 -d'.')
+    if echo $PWD | grep -o -q $rwsdir; then wsname=$rwsname && return; fi
+  done <<< "$(echo "$(listall)")"
+  # when here then not in workspace dir
+  echo "error: you are not in a workspace directory. your registered workspaces are:" && wslist="$(echo "$(listall)")" && echo "${wslist:-'<no workspaces defined yet>'}" && exit 1
+}
+
+# helper to check number of arguments
+function allowedargcount () {
+  ( test $initialcount -ne $1 ) && ( usage && echo 1>&2 "error: wrong number of arguments" && exit 1 ) 
+}
+
+# execute the bulk operation
+function executBulkOp () {
+  checkGitCommand
+  if ! $allwsmode && ! $singlemode; then wsnameToCurrent; fi # by default git bulk works within the 'current' workspace
+  listall | while read workspacespec; do
+    cwsname=$(echo $workspacespec | cut -f1 -d' ' | cut -f2 -d'.')
+    if [[ -n $wsname ]] && [[ $cwsname != $wsname ]]; then continue; fi
+    wslocation=$(echo $workspacespec | cut -f2 -d' ')
+    eval cd "$wslocation"
+    actual=$(pwd)
+    echo "Executing bulk operation in workspace ${invers}$actual${reset}"
+    eval find . -name ".git" | while read line; do
+      gitrepodir=${line::${#line}-5} # cut the .git part of find results to have the root git directory of that repository
+      eval cd "$gitrepodir" # into git repo location
+      curdir=$(pwd)
+      echo "Current repository: ${actual#${curdir%/*}}/${bldred}${curdir##*/}${reset}"
+      guardedExecution
+      eval cd "$wslocation" # back to origin location of last find command
+    done 
+  done 
+}
+
+initialcount="${#}"
```

Comment:
IMHO, `initalcount` is not a good variable name...
People don't know what it counts for at the first glance.